test/unit/org/apache/cassandra/index/sai/utils/PostingListTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.utils;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+import org.apache.cassandra.index.sai.disk.PostingList;
+
+import static org.junit.Assert.assertEquals;
+
+public class PostingListTest extends SAIRandomizedTester
+{
+    @Test
+    public void testPeekablePostingList() throws IOException
+    {
+        PostingList.PeekablePostingList postingList = new ArrayPostingList(new int[]{ 1, 2, 3 }).peekable();
+        assertEquals(3, postingList.size());
+        assertEquals(1, postingList.peek());
+        assertEquals(1, postingList.nextPosting());
+        assertEquals(2, postingList.peek());
+        assertEquals(2, postingList.nextPosting());
+        assertEquals(3, postingList.peek());
+        assertEquals(3, postingList.nextPosting());
+        assertEquals(PostingList.END_OF_STREAM, postingList.peek());
+        assertEquals(PostingList.END_OF_STREAM, postingList.nextPosting());
+
```

Comment:
nit: No setup cost, so maybe split this into 3 tests, like `testPeekableNextPosting()`, `testPeekableAdvance()`, `testPeekableAdvanceWithoutConsuming()` (names could change if `PeekablePostingList` gets its own top-level class I guess)